pkg/controller/dataimportcron-controller.go

```diff
@@ -702,10 +711,19 @@ func addDataImportCronControllerWatches(mgr manager.Manager, c controller.Contro
 	return nil
 }
 
-func (r *DataImportCronReconciler) cronJobExists(ctx context.Context, cron *cdiv1.DataImportCron) bool {
+func (r *DataImportCronReconciler) cronJobExistsAndUpdated(ctx context.Context, cron *cdiv1.DataImportCron) (bool, error) {
 	var cronJob batchv1.CronJob
 	cronJobNamespacedName := types.NamespacedName{Namespace: r.cdiNamespace, Name: GetCronJobName(cron)}
-	return r.uncachedClient.Get(ctx, cronJobNamespacedName, &cronJob) == nil
```

Comment:
Why were we doing uncached client here btw?

Reply:
afair r.client.Get() sometime returned NotFound after creating CronJob. However, I'll try again r.client for Get() and Update() here and check if it's consistent.

Reply:
Removed. So far no surprises.